tooling/api/src/window.ts

```diff
@@ -1517,6 +1472,238 @@ class WindowManager extends WebviewWindowHandle {
       }
     })
   }
+
+  // Listeners
+
+  /**
+   * Listen to window resize.
+   * @example
+   * ```typescript
+   * import { appWindow } from "@tauri-apps/api/window";
+   * appWindow.onResized(({ payload: size }) => {
+   *  console.log('Window resized', size);
+   * });
+   * ```
+   *
+   * @param handler
+   * @returns A promise resolving to a function to unlisten to the event.
+   */
+  async onResized(handler: EventCallback<PhysicalSize>): Promise<UnlistenFn> {
+    return this.listen<PhysicalSize>('tauri://resize', handler)
+  }
+
+  /**
+   * Listen to window move.
+   * @example
+   * ```typescript
+   * import { appWindow } from "@tauri-apps/api/window";
+   * appWindow.onMoved(({ payload: position }) => {
+   *  console.log('Window moved', position);
+   * });
+   * ```
+   *
+   * @param handler
+   * @returns A promise resolving to a function to unlisten to the event.
+   */
+  async onMoved(handler: EventCallback<PhysicalPosition>): Promise<UnlistenFn> {
+    return this.listen<PhysicalPosition>('tauri://move', handler)
+  }
+
+  /**
+   * Listen to window close requested. Emitted when the user requests to closes the window.
+   * @example
+   * ```typescript
+   * import { appWindow } from "@tauri-apps/api/window";
+   * import { confirm } from '@tauri-apps/api/dialog';
+   * appWindow.onCloseRequested(async (event) => {
+   *   const confirmed = await confirm('Are you sure?');
+   *   if (!confirmed) {
+   *     // user did not confirm closing the window; let's prevent it
+   *     event.preventDefault();
+   *   }
+   * });
+   * ```
+   *
+   * @param handler
+   * @returns A promise resolving to a function to unlisten to the event.
+   */
+  async onCloseRequested(
+    handler: (event: CloseRequestedEvent) => void
```

Comment:
```suggestion
    handler: EventCallback<CloseRequestedEvent>
```

Reply:
These are not the same, `EventCallback<CloseRequestedEvent>` translates to `(event: Event<CloseRequestedEvent>) => void` and we're wrapping the event fields in the CloseRequestedEvent directly (just to add the preventDefault method).